File: examples/python/core/logger/automatic-context-enrichment.py
@durable_execution
def handler(event: dict, context: DurableContext) -> str:
    # This log includes: execution_arn
    context.logger.info("Top-level log")
    
    result: str = context.step(
        lambda _: "processed",
        name="process_data",
    )
    
    # This log includes: execution_arn, parent_id, name, attempt
    context.logger.info("Step completed")
    
    return result

File: examples/java/core/logger/automatic-context-enrichment.java
public class AutoContextHandler extends DurableHandler<Map<String, Object>, String> {

    @Override
    public String handleRequest(Map<String, Object> event, DurableContext ctx) {
        // This log includes: durableExecutionArn
        ctx.getLogger().info("Top-level log");

        String result = ctx.step("process_data", String.class, stepCtx -> {
            // This log includes: durableExecutionArn, operationId, operationName, attempt
            stepCtx.getLogger().info("Step completed");
            return "processed";
        });

        return result;
    }
}
